nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -137,7 +138,11 @@ protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
     if (pti.reference() != null) {
       newReference = loadReference(pti.reference());
     }
-    return new NessieTableOperations(NessieUtil.toKey(pti.tableIdentifier()), newReference, client, fileIO);
+    NessieTableOperations tableOperations =
+        new NessieTableOperations(NessieUtil.toKey(pti.tableIdentifier()), newReference, client, fileIO);
+    // TODO: is there a better way to pass the catalog options to the TableOperations than this?
```

Comment:
Add it to the constructor? The constructor is public, but I don't think there is a need for it to be public. You could create a package-private one that is called here instead. (Maybe @rymurr can also comment)

Reply:
for some reason I assumed that the signature of the constructor was defined and shouldn't be changed. Definitely better to pass this via the constructor

Reply:
@rymurr, is there a reason why the constructor was public?

Reply:
I think it was public because it was created before `CatalogUtil.loadCatalog`. No real reason for it to be public now.

Reply:
ahhh excuse me, you meant the TableOps. That was just a mistake I suspect. Even less of a reason for that to be public.